### ISSUE TITLE
SurfaceGUI Bug - Appearing Raised Dependent on Position

### ISSUE THREAD

**Issue by DeveloperBlue** [Anaminus/roblox-bug-tracker#630]
Say you have a block with a SurfaceGUI on it. Dependent on the position, the SurfaceGUI seems to render itself farther and farther away, off the block's face.

This can be replicated using both the X,Y, and/or Z axis.

At Vector3.new(0,0,0), the SurfaceGUI fits snugly and flatly onto it's respective surface.
At Vector3.new(0,0,1000), the SurfaceGUI is elevated off of it's surface by about .4 of a stud.
At Vector3.new(0,0,8000), the SurfaceGUI is elevated off of it's surface by like 2-3 studs. 

I was able to replicate this on two separate devices in studio and in-game with multiple accounts.

The Positions were set with both a) the Studio Dragger or b) the Properties panel.
When clicking on a textbutton on the SurfaceGUI, you aim for where the button SHOULD be rendered, NOT where it actually appears to be (4 studs in the air). The client recognizes that the buttons are there, it's just drawn incorrectly?

![replication of surfacegui bug](https://cloud.githubusercontent.com/assets/11480933/7900464/775ff74e-0727-11e5-82bb-331fc231547a.png)

**EDIT:**
According to Merely, "This was a change by maxvee to the client to fight z-fighting, but it caused this side effect."

~~Although it should be noted that while Merely says "client", the bug is visible both studio's edit mode, server-sided, and client-sided.~~

![zfighting](https://cloud.githubusercontent.com/assets/11480933/7925383/dca2aa1e-0891-11e5-8a77-fa79a15d1d74.PNG)

**EDIT:**
Noted by Vectricity, the 'gap' actively increases the further the Part containing the SurfaceGUI move away from 0,0,0. 

*Ex, placing a BodyForce inside a part with a SurfaceGUI and shooting it off into any logical direction, you can witness the SurfaceGUI render itself further and further away from it's adorned part in the direction of it's respective face.*

**Comment by MrJimenex** [comment]
ROBLOX announced that the bug will be fixed on Wednesday night.

**Comment by DeveloperBlue** [comment]
Can I ask for your source on this? It's Thursday and I can still repro the bug.

![bug repro2](https://cloud.githubusercontent.com/assets/11480933/7991318/55958254-0ac5-11e5-9d54-df279d8c575e.png)

**Comment by DigiTechs** [comment]
> Although it should be noted that while Merely says "client", the bug is visible both studio's edit mode, server-sided, and client-sided.

That's due to the studio server also rendering it. To the server (on a real game server, i.e. online), the SurfaceGui isn't rendered at all so it does not have a rendered position in-game. That's what he means by 'client'.

**Comment by Vectricity** [comment]
I've been suffering from this issue for a long time, so I'm glad to see it being reported here. Also, I had no idea it involved the position of the part. It makes sense, given that my place has content scattered thousands of studs away from point 0,0,0. I move the brick with the SurfaceGui to compensate, but that causes its own problems. Hopefully this gets fixed at some point as it is quite the annoying bug.

**Comment by Vectricity** [comment]
I submitted a pull request for this. Honestly one of the most difficult things to figure out.

**Comment by DigiTechs** [comment]
It isn't really that hard to reproduce. Just place a part far from the origin and put a SurfaceGui on it. Also, it's worth pointing out that a few fixes went out for this recently. (from what I remember looking at the ROBLOX Developer forums, anyway)

**Comment by Vectricity** [comment]
I checked just yesterday, and it still occurred (unless you meant in a testing environment). Moreover, your proposed test doesn't fully flush out the problem. I witnessed that the SurfaceGui actively gets further from the origin block as you move away from 0,0,0. That is additional detail that was not present. Also, it shows that there is something purposefully and constantly changing its distance. That said, I'll wait and see whether the issue is fixed soon.
